neuraxle/data_container.py

```diff
@@ -0,0 +1,242 @@
+        conv_data_inputs = convolved_1d(stride=stride, iterable=self.data_inputs, kernel_size=kernel_size)
+        conv_expected_outputs = convolved_1d(stride=stride, iterable=self.expected_outputs, kernel_size=kernel_size)
+
+        for current_ids, data_inputs, expected_outputs in zip(conv_current_ids, conv_data_inputs,
+                                                              conv_expected_outputs):
+            yield DataContainer(
+                current_ids=current_ids,
+                data_inputs=data_inputs,
+                expected_outputs=expected_outputs
+            )
+
+    def __iter__(self):
+        """
+        Iter method returns a zip of all of the current_ids, data_inputs, and expected_outputs in the data container.
+
+        :return: iterator of tuples containing current_ids, data_inputs, and expected outputs
+        :rtype: Iterator[Tuple]
+        """
+        current_ids = self.current_ids
+        if self.current_ids is None:
+            current_ids = [None] * len(self.data_inputs)
+
+        expected_outputs = self.expected_outputs
+        if self.expected_outputs is None:
+            expected_outputs = [None] * len(self.data_inputs)
+
+        return zip(current_ids, self.data_inputs, expected_outputs)
+
+    def __repr__(self):
+        return str(self)
+
+    def __str__(self):
+        return self.__class__.__name__ + "(current_ids=" + repr(list(self.current_ids)) + ", ...)"
+
+    def __len__(self):
+        return len(self.data_inputs)
+
+
+class ExpandedDataContainer(DataContainer):
+    """
+    Sub class of DataContainer to expand data container dimension.
+
+    .. seealso::
+        :class:`ExpandedDataContainer`,
+    """
+    def __init__(self, current_ids, data_inputs, expected_outputs, old_current_ids):
+        DataContainer.__init__(
+            self,
+            current_ids,
+            data_inputs,
+            expected_outputs
+        )
+
+        self.old_current_ids = old_current_ids
+
+    def reduce_dim(self) -> 'DataContainer':
+        """
+        Reduce DataContainer to its original shape with a list of multiple current_ids, data_inputs, and expected outputs.
+
+        :return: reduced data container
+        :rtype: DataContainer
+        """
+        return DataContainer(
+            current_ids=self.old_current_ids,
+            data_inputs=self.data_inputs[0],
+            expected_outputs=self.expected_outputs[0],
+        )
+
+    @staticmethod
+    def create_from(data_container: DataContainer) -> 'ExpandedDataContainer':
+        """
+        Create ExpandedDataContainer with the given summary hash for the single current id.
+
+        :param data_container: data container to transform
+        :type data_container: DataContainer
+        :return: expanded data container
+        :rtype: ExpandedDataContainer
+        """
+        return ExpandedDataContainer(
+            current_ids=[data_container.summary_hash()],
```

Comment:
please just note that if we'd have kept the hashers' `summary_hash` and that in base_step too, then we'd have needed data_conatiner.summary_hash to be named differently to avoid confusion. Because here it's a getter and it's a **different** method. 

^ The comment I made above is recurrent in many PRs. Thinking of the `.save` prs especially.

Reply:
Thinknig twice, I think we need to rename `summary_hash` to `summarize_hashes`. It is truly a kind of summary of all the hashes, and not a hash of a summary.